.pre-commit-config.yaml

```diff
@@ -0,0 +1,50 @@
+default_language_version:
+  python: python3
+exclude: |
+  (?x)^(
+  integration_tests/test_cases_bloom560m.yaml|
+  integration_tests/test_cases_common.yaml|
+  integration_tests/test_cases_gpt2.yaml|
+  integration_tests/test_cases_mt0_ost.yaml|
+  integration_tests/test_cases_mt0small.yaml|
+  integration_tests/test_cases_tinyllama.yaml|
+  integration_tests/test_cases_tinystarcoderpy.yaml
+  )$
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.5.0
+    hooks:
+      - id: check-added-large-files
+      - id: check-case-conflict
+      - id: check-docstring-first
+      - id: check-executables-have-shebangs
+      - id: check-json
+      - id: check-merge-conflict
+        args: ["--assume-in-merge"]
+      - id: check-toml
+      - id: check-yaml
+      - id: end-of-file-fixer
+      - id: mixed-line-ending
+        args: ["--fix=lf"]
+      - id: sort-simple-yaml
+      - id: trailing-whitespace
+  - repo: https://github.com/doublify/pre-commit-rust
+    rev: v1.0
+    hooks:
+      - id: fmt
+      - id: cargo-check
+      # - id: clippy # cargo clippy currently fails
```

Comment:
@njhill @ckadner  we could also enable the  `cargo clippy` hook, although I fear it could be slow and annoying. I still feel it would be a valuable addition, maybe we could maybe add it as part of CI.

Please note that these hooks are run as part of CI, since [this line](https://github.com/IBM/text-generation-inference/pull/26/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88R31) was added in this PR in the test workflow

Reply:
See https://github.com/IBM/text-generation-inference/pull/28 for `ruff` code linting